File: chapter09/sample.scala
// 生成0到99的整型数组
val arr = (0 until 100).toArray
// 使用parallelize生成RDD
val rdd = sc.parallelize(arr)
 
// 不带seed，每次采样结果都不同
rdd.sample(false, 0.1).collect
// 结果集：Array(11, 13, 14, 39, 43, 63, 73, 78, 83, 88, 89, 90)
rdd.sample(false, 0.1).collect
// 结果集：Array(6, 9, 10, 11, 17, 36, 44, 53, 73, 74, 79, 97, 99)
 
// 带seed，每次采样结果都一样
rdd.sample(false, 0.1, 123).collect
// 结果集：Array(3, 11, 26, 59, 82, 89, 96, 99)
rdd.sample(false, 0.1, 123).collect
// 结果集：Array(3, 11, 26, 59, 82, 89, 96, 99)
 
// 有放回采样，采样结果可能包含重复值
rdd.sample(true, 0.1, 456).collect
// 结果集：Array(7, 11, 11, 23, 26, 26, 33, 41, 57, 74, 96)
rdd.sample(true, 0.1, 456).collect
// 结果集：Array(7, 11, 11, 23, 26, 26, 33, 41, 57, 74, 96)

File: basic_spark/chapter09/sample.py
import random
from pyspark import SparkContext

sc = SparkContext()

arr = [random.randint(0, 99) for _ in range(100)]
rdd = sc.parallelize(arr)

l1 = rdd.sample(False, 0.1).collect()
print(l1)

l2 = rdd.sample(False, 0.1).collect()
print(l2)

l3 = rdd.sample(False, 0.1, 123).collect()
print(l3)

l4 = rdd.sample(False, 0.1, 123).collect()
print(l4)

l5 = rdd.sample(True, 0.1, 456).collect()
print(l5)

l6 = rdd.sample(True, 0.1, 456).collect()
print(l6)
